cdap-ranger/cdap-ranger-binding/src/test/java/co/cask/cdap/security/authorization/ranger/binding/RangerAuthorizerTest.java

```diff
@@ -0,0 +1,178 @@
+import co.cask.cdap.proto.security.Principal;
+import co.cask.cdap.security.spi.authorization.UnauthorizedException;
+import com.google.common.collect.ImmutableSet;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Test {@link RangerAuthorizer} through a policies stored in a file under resources/cdap-policies.json
+ */
+public class RangerAuthorizerTest {
+
+  private static final NamespaceId NAMESPACE_DEFAULT = new NamespaceId("default");
+  private static final Principal ALI = new Principal("ali", Principal.PrincipalType.USER);
+  private static final Principal SAGAR = new Principal("sagar", Principal.PrincipalType.USER);
+  private static final Principal SHANKAR = new Principal("shankar", Principal.PrincipalType.USER);
+  private static final Principal MATT = new Principal("matt", Principal.PrincipalType.USER);
+  private static final Principal POORNA = new Principal("poorna", Principal.PrincipalType.USER);
+  private static final Principal RSINHA = new Principal("rsinha", Principal.PrincipalType.USER);
+  private static final Principal DEREK = new Principal("derek", Principal.PrincipalType.USER);
+  private static final Principal ALBERT = new Principal("albert", Principal.PrincipalType.USER);
+  private static final Principal SREE = new Principal("sree", Principal.PrincipalType.USER);
+  private static final Principal YAOJIE = new Principal("yaojie", Principal.PrincipalType.USER);
+
+  private static RangerAuthorizer authorizer;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    authorizer = new RangerAuthorizer();
+    authorizer.initialize(new InMemoryAuthorizationContext());
+  }
+
+  @Test
+  public void testCompletePrivileges() throws Exception {
+    // Test Privileges for ali who has ADMIN on stream:default:teststream
+    authorizer.enforce(NAMESPACE_DEFAULT.stream("teststream"), ALI, Action.ADMIN);
+    Assert.assertEquals(ImmutableSet.of(NAMESPACE_DEFAULT, NAMESPACE_DEFAULT.stream("teststream")),
+                        authorizer.isVisible(ImmutableSet.<EntityId>of(NAMESPACE_DEFAULT,
+                                                                       NAMESPACE_DEFAULT.stream("teststream")), ALI));
+    testEnforceFail(NAMESPACE_DEFAULT.stream("anotherstream"), ALI, Action.ADMIN);
+    testVisibleFail(ImmutableSet.<EntityId>of(new NamespaceId("anothernamespace")), ALI);
+    testEnforceFail(NAMESPACE_DEFAULT.stream("teststream"), ALI, Action.READ);
+    testEnforceFail(NAMESPACE_DEFAULT, ALI, Action.ADMIN);
+
+    // Test privileges for rsinha who has ADMIN on namespace:default
+    authorizer.enforce(NAMESPACE_DEFAULT, RSINHA, Action.ADMIN);
+    testEnforceFail(new NamespaceId("anothernamespace"), RSINHA, Action.ADMIN);
+    Assert.assertEquals(ImmutableSet.of(NAMESPACE_DEFAULT),
+                        authorizer.isVisible(ImmutableSet.<EntityId>of(NAMESPACE_DEFAULT), RSINHA));
```

Comment:
Can you put a stream id here to test we can view the namespace but not the stream?